### DISCUSSION TITLE
Edge does not accept Manifest v3 icons format

### DISCUSSION THREAD

**Discussion by michaelaflores** [microsoft/MicrosoftEdge-Extensions#73] [Discussions]
In Chrome, under Manifest v3 the icons key is expected to be an array according to a warning in Chrome. If you try to submit an extension using this format to Edge Add-ons, you'll get the following error:

```
Package acceptance validation error: The JSON token is invalid. Expected Object but got Array. Error code: icons: [ { "src": "/assets/icons/favicon-16x16.png", "type": "image/png", "sizes": "16x16" }, { "src": "/assets/icons/favicon-32x32.png", "type": "image/png", "sizes": "32x32" }, { "src": "/assets/icons/favicon.ico", "type": "image/ico", "sizes": "48x48" } ] Line: 12 Column: 12
```

Are there plans to support this format? In the mean time, is it expected that developers will maintain two versions of their manifest to support Edge?

Granted Chrome itself won't load a manifest with icons value as an array so it's possible this warning was a bug.

**Comment by sivsouvamMSFT** [comment]
Hi @michaelaflores, thank you for bringing this up. We will look into this & get back to you soon on it.

**Comment by Bhuvana-Priya** [comment]
Hi @michaelaflores , our engineering teams have reviewed this and I wanted to let you know that as per Manifest V3 file format, the "icons" field should be an object, and not array for extensions. Hence, the error you have mentioned "Expected object but got array" seems valid. Please set the "icons" field as an object.

Reference for how to use the icons field can be found here: [Chrome Extensions Manifest: icons - Chrome Developers](https://developer.chrome.com/docs/extensions/mv3/manifest/icons/)
